common/src/main/java/io/pravega/common/SecurityConfigDefaults.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common;
```

Comment:
`common` does not seem to be the right place for these definitions. `common` is more about classes that can be used across components of Pravega and even outside the project. these definitions here are fairly specific to Pravega, so I'd say it should be in `shared` instead.

Reply:
Sure, thanks for pointing that out. I read the readme file there and noticed that idea as well, but got misled by a couple of other examples. 

I tried moving it to the `shared` > `cluster` subproject, but moving there would cause a series of build changes. After reflecting a bit and looking around for other test utilities/helpers, I felt that the best home for it is in `pravega` > `test` > `testcommon` Gradle subproject in the package `io.pravega.test.common`. I've made that change in the latest commit. Hope that makes sense.

Reply:
What about `io.pravega.shared.test`? We don't have a `test` package under `shared`, but we could create one in this PR and it would make more sense to me because of the following:

- `shared` is supposed to contain classes that are pravda-specific and shared across components.
- `test.common` is borderline ok for this. Even though we are using standard security primitives, I don't see these constants being useful to anyone outside pravega. To determine whether goes into `common` or `test.common`, I ask myself: could anyone be interested in using this outside the context of pravega?

Reply:
Given the purpose of the `pravega/shared` Gradle subproject, it makes sense to have a `pravega/shared/test` sub-project with the classes under `io.pravega.shared.test` package/namespace. 

That being said, all the classes that are used across tests in other Gradle sub-projects lie inside of the `pravega/test/testcommon` subproject today inside the namespace `io.pravega.test.common`. Witness the `io.pravega.test.common.AssertExtensions` class that lies in the same package as `io.pravega.test.common.SecurityConfigDefaults` class that I have introduced. This scheme also makes sense to me, as all test related stuff is cohesively located in one place. Arguably, this scheme is even better as it favors functional cohesion over logical cohesion. 

I think we move `SecurityConfigDefaults` elsewhere, it'd make sense to move other existing classes in testcommon, as well (with consequential build script changes as well). Doing so will take the scope of this PR way well beyond the original intention. 

Therefore, I'd like to propose we keep it as is right now. What do you say?

Reply:
`io.pravega.test.common.AssertExtensions` is something that I see others outside Pravega benefitting from. `test/common` is the right place for it.

I can't recall any class for tests like `io.pravega.test.common.SecurityConfigDefaults`, so I'm tempted to say that it is the first time we are encountering this situation.

But, if you tell me that we need deeper changes than just moving the file to have those definitions in `shared/test`, then I can be convinced to postpone the reorganization. Otherwise, I'd rather avoid the tech debt.

Reply:
A lot of tests across several Gradle subprojects use the `SecurityConfigDefaults`. There will be deeper changes required to move it (like Gradle build changes, etc.), and therefore I'd request that we postpone this reorganization.